itk_cucim/filtering/smoothing.py

```diff
@@ -0,0 +1,40 @@
+"""cuCIM accelerated filters for the ITKSmoothing module."""
+
+import itk
+import cucim.skimage
+import numpy as np
+import cupy as cp
+
+def cucim_median_image_filter(*args, **kwargs):
+    input_image = args[0]
+    ref_filt = itk.MedianImageFilter.New(*args, **kwargs)
+    wrapper = itk.PyImageFilter.New(input_image)
```

Comment:
Not sure if I have the ITK API usage right here, but do we need to explicitly set the filter to use a single thread on the host side? (I don't think we want to launch multiple GPU kernels in parallel).

```suggestion
    threader = itk.MultiThreaderBase.New()
    threader.SetMaximumNumberOfThreads(1)
    wrapper = itk.PyImageFilter.New(x, MultiThreader=threader)
```

Reply:
Launching multiple GPU kernels is good to think about, but `generate_output_information` and `generate_data` will be executed in a single thread, so we do not have to worry about launching multiple GPU kernels in parallel.

I am curious -- is there a way for cucim.skimage.filters.median or the underlying functionality to launch multiple GPU kernels? Is there settings on the `cucim` package that would control this?

Reply:
Great, looks like we don't need this suggestion then. 

You can launch multiple kernels from [different streams](https://docs.cupy.dev/en/v11.0.0b2/user_guide/cuda_api.html). We are not using that anywhere within `cucim.skimage` at the moment. As in CuPy itself, we do try to avoid things that require device synchronization, but it is not possible for every function.

For multi-GPU, there may be a handful of things in CuPy that support this ([FFT](https://docs.cupy.dev/en/stable/user_guide/fft.html#multi-gpu-fft) is the one I'm aware of), but mostly one would likely use Dask's `map_blocks` or similar to distribute computation across multiple GPUs.